### DISCUSSION TITLE
Should source-built .NET SDK include testhost.exe by default

### DISCUSSION THREAD

**Discussion by xgqt** [dotnet/source-build#4095] [Q&A]
Hello!

One of Gentoo users has run onto the issue where our build on .NET SDK did not include the testhost.exe file.
https://bugs.gentoo.org/923832

I have some questions that might help us sort it out:
- Is this a expected behavior that should happen with default build?
- Is this file available only when mono build is requested? 
- What is the setup of environment and flags to build it while building .NET SDK?

**Comment by omajid** [comment]
https://github.com/dotnet/installer/blob/main/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/baselines/MsftToSbSdkFiles.diff - which tracks the difference between Microsoft-built and source-built .NET SDKs - points out that the missing `TestHostNetFramework` directory is a known issue.

Looks like this was added as part of https://github.com/dotnet/installer/pull/14580.

I *think* the reasoning was that .NET Framework bits are not usable/useful on Linux. However, if we have users and projects that depend on this, we might want to re-enable this.

@MichaelSimons should we convert this to an issue and investigate removing the `TestHostNetFramework` diff between source-build SDK and Microsoft-built SDK?

**Reply by MichaelSimons** [reply]
Yes I will convert this to an issue to investigate.  It is not clear to me at the moment if this is something we can support.

**Reply by MichaelSimons** [reply]
See https://github.com/dotnet/source-build/issues/4099
